src/community/mutual_information.jl

```diff
@@ -0,0 +1,67 @@
+function entropy(membership::Vector{Int})
+    n = length(membership)
+    nb_comms = maximum(membership)
+    groups = Array(Set{Int}, nb_comms)
+    for i=1:nb_comms
+        groups[i] = Set{Int}()
+    end
+    for i=1:n
+        push!(groups[membership[i]], i)
+    end
+    h = 0.0
+    for grp in groups
+        n_c = length(grp)
+        h += -(n_c/n)log(n_c/n)
+    end
+    h
+end
```

Comment:
Could you rewrite this function without allocating Sets but just using counters?
  ```julia
groups = zeros(Int, nb_comms)
for i=1:n
        groups[membership[i]] += 1
 end
```